feat(recipe-list): bare-bones recipe list

marcin-hoa opened the pull request:
### Changes
- added `proxy.config.json` and included it in hoa-recipes's project.json
- added libs
    - feature-recipe-list
    - data-access-api and needed DTOs
- updated prettier and fixed prettier config

### Questions
- app.component.html now has two imports from libs folder. It will grow over time. Should it also migrate to libs?

marcin-hoa committed:
feat(recipe-list): bare-bones recipe list

DDonochVA commented on libs/data-access-api/src/lib/data-access-api/recipes.repository.ts:
```diff
@@ -0,0 +1,17 @@
+
+  constructor(private client: HttpClient) {
+    this.baseUrl = `/api/recipes`;
+  }
```
give the new `inject` function a try
```suggestion
  private baseUrl = `/api/recipes`;
  private client = inject(HttpClient);
```

DDonochVA commented on libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts:
```diff
@@ -0,0 +1,25 @@
+import { MatListModule } from '@angular/material/list';
+import { RecipeDto, RecipesRepository } from '@hoa-recipes/data-access-api';
+
+@Component({
```
I would add `OnPush` strategy

DDonochVA commented on apps/hoa-recipes/src/app/app.component.ts:
```diff
@@ -1,11 +1,12 @@
+import { FeatureRecipeListComponent } from '@hoa-recipes/feature-recipe-list';
 import { FeatureTitleBarComponent } from '@hoa-recipes/feature-title-bar';
 
 @Component({
```
I would add `OnPush` strategy

DDonochVA commented on libs/feature-title-bar/src/lib/feature-title-bar/feature-title-bar.component.ts:
```diff
@@ -5,6 +5,7 @@ import { MatDialog } from '@angular/material/dialog';
 import { UiAuthorDialogComponent } from '@hoa-recipes/ui-author-dialog';
+import { take } from 'rxjs';
 
 @Component({
```
OnPush

DDonochVA replied:
+make sure if `CommonModule` is necessary

DDonochVA commented on libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts:
```diff
@@ -0,0 +1,25 @@
+@Component({
+  selector: 'hoa-recipes-feature-recipe-list',
+  standalone: true,
+  imports: [CommonModule, MatListModule, MatIcon],
```
make sure if `CommonModule` is necessary

DDonochVA commented on apps/hoa-recipes/src/app/app.component.html:
```diff
@@ -1,2 +1,3 @@
 <hoa-recipes-feature-title-bar></hoa-recipes-feature-title-bar>
+<hoa-recipes-feature-recipe-list></hoa-recipes-feature-recipe-list>
```
you can use self-closed tags
```suggestion
<hoa-recipes-feature-recipe-list />
```

DDonochVA commented on libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts:
```diff
@@ -0,0 +1,25 @@
+  constructor(private recipeRepository: RecipesRepository) {}
+
+  ngOnInit(): void {
+    this.recipeRepository.getAll().subscribe((res) => {
```
I would recommend managing data with NgRX

DDonochVA reviewed (commented):
Summary:
- Use `OnPush` strategy in all components
- Injecting can be done via `inject` function instead of constructor
- Avoid importing CommonModule as it was broke up into smaller chunks (NgIf, NgFor)
- Install NgRX

marcin-hoa committed:
feat: renamend feature-title-bar to ui-title-bar

marcin-hoa committed:
feat: added json-server to mock backend

marcin-hoa committed:
feat: added OnPush strategy to all components

marcin-hoa committed:
feat: added self closing tags

marcin-hoa commented:
I fixed almost all things. The only one left is NgRx, which I'm gonna try out now. 

I also added `json-server` to replace crudcrud. It works basically the same, but it is unlimited :)

marcin-hoa committed:
fix: ngrx store

marcin-hoa committed:
redactor: moved ngrx state to data-acces-recipes lib

marcin-hoa committed:
refactor: minor routing changes and styling fixes
